project-geometry/src/main/java/com/esri/arcgismaps/sample/projectgeometry/MainActivity.kt

```diff
@@ -0,0 +1,158 @@
+                width = 30f
+                height = 30f
+                // offset in +y axis so the marker spawned
+                // is right on the touch point
+                offsetY = 25f
+            }
+        }
+    }
+
+    // creates a graphic overlay
+    private val graphicsOverlay: GraphicsOverlay = GraphicsOverlay()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        // authentication with an API key or named user is
+        // required to access basemaps and other location services
+        ArcGISEnvironment.apiKey = ApiKey.create(BuildConfig.API_KEY)
+        lifecycle.addObserver(mapView)
+
+        // configure mapView assignments
+        mapView.apply {
+            // create and add a map with a navigation night basemap style
+            map = ArcGISMap(BasemapStyle.ArcGISNavigationNight)
+            // add the graphics overlay to display marker graphics
+            graphicsOverlays.add(graphicsOverlay)
+            // set the default viewpoint to Redlands,CA
+            setViewpoint(Viewpoint(34.058, -117.195, 5e4))
+        }
+
+        lifecycleScope.launch {
+            // check if the map has loaded successfully
+            mapView.map?.load()?.onSuccess {
+                // capture and collect when the user taps on the screen
+                mapView.onSingleTapConfirmed.collect { event ->
+                    event.mapPoint?.let { point -> geoViewTapped(point) }
+                }
+            }?.onFailure {
+                // if map load failed, show the error
+                showError("Error Loading Map", mapView)
+            }
+        }
+    }
+
+    /**
+     * Handles the SingleTapEvent by drawing a marker and performs a Spatial reference transformation
+     * of the tapped Location using GeometryEngine and displays the result
+     */
+    private fun geoViewTapped(point: Point) {
+        // show the marker where the user tapped on the map
+        graphicsOverlay.graphics.apply {
+            // this removes previously added markers
+            clear()
+            // add a new marker graphic
+            add(markerGraphic.apply {
+                // updates the geometry location to the current tap
+                geometry = point
+            })
+        }
+        // project the web mercator location into a WGS84
+        val projectedPoint = GeometryEngine.project(point, SpatialReference.wgs84())
+        // build and display the projection result as a string
+        infoTextView.text = getString(
+            R.string.projection_info_text,
+            point.toDisplayFormat(),
+            projectedPoint.toDisplayFormat()
+        )
+    }
+
+    /**
+     * Extension function for the Point type that returns
+     * a float-precision formatted string suitable for display
+     */
+    private fun Point.toDisplayFormat() =
+        "${String.format("%.5f", x)}, ${String.format("%.5f", y)}"
```

Comment:
Good idea to use an extension function here :+1: Although I would suggest moving it because extension functions should generally go outside of class scope, as top-level functions in the file. This is recommended because otherwise there are two `this` references in the body of the function: in this case, the one for Point and the one for MainActivity. Since the function only uses the one for Point it's clearer to put the function outside MainActivity to show that it doesn't depend on MainActivity's state. Usually they're placed at the bottom of the file and still marked with `private`.